tests/test_inner.py

```diff
@@ -848,3 +867,59 @@ def test_workflow_plugin_results():
     assert workflow.postbuild_results == {'post_build_value': 'post_build_value_result'}
     assert workflow.prepub_results == {'pre_publish_value': 'pre_publish_value_result'}
     assert workflow.exit_results == {'exit_value': 'exit_value_result'}
+
+
+def test_cancel_build(request):
+    """
+    Verifies that exit plugins are executed when the build is cancelled
+    """
+    this_file = inspect.getfile(PreRaises)
+    mock_docker()
+    fake_builder = MockInsideBuilder()
+    flexmock(InsideBuilder).new_instances(fake_builder)
+
+    watch_pre = Watcher()
+    watch_prepub = Watcher()
+    watch_post = WatcherWithPause()
+    watch_exit = Watcher()
+
+    fake_logger = FakeLogger()
+    existing_logger = atomic_reactor.plugin.logger
+
+    def restore_logger():
+        atomic_reactor.plugin.logger = existing_logger
+
+    request.addfinalizer(restore_logger)
+    atomic_reactor.plugin.logger = fake_logger
+
+    workflow = DockerBuildWorkflow(MOCK_SOURCE, 'test-image',
+                                   prebuild_plugins=[{'name': 'pre_watched',
+                                                      'args': {
+                                                          'watcher': watch_pre
+                                                      }}],
+                                   prepublish_plugins=[{'name': 'prepub_watched',
+                                                        'args': {
+                                                            'watcher': watch_prepub,
+                                                        }}],
+                                   postbuild_plugins=[{'name': 'post_watched_raises',
```

Comment:
Ah, its a remainder from the first attempt - I tried to make it fail. Replacing with `post_watched`